pandas/io/excel/_calaminereader.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+from datetime import (
+    date,
+    datetime,
+    time,
+)
+from tempfile import NamedTemporaryFile
+from typing import Union
+
+from pandas._typing import (
+    FilePath,
+    ReadBuffer,
+    Scalar,
+    StorageOptions,
+)
+from pandas.compat._optional import import_optional_dependency
+
+import pandas as pd
+
+from pandas.io.common import stringify_path
+from pandas.io.excel._base import (
+    BaseExcelReader,
+    inspect_excel_format,
+)
+
+ValueT = Union[int, float, str, bool, time, date, datetime]
+
+
+class __calamine__:
+    pass
+
+
+class CalamineExcelReader(BaseExcelReader):
+    book: str
+    _sheet_names: list[str] | None = None
+
+    def __init__(
+        self,
+        filepath_or_buffer: FilePath | ReadBuffer[bytes],
+        storage_options: StorageOptions = None,
+    ) -> None:
+        import_optional_dependency("python_calamine")
+        super().__init__(filepath_or_buffer, storage_options=storage_options)
+
+    @property
+    def _workbook_class(self) -> type[__calamine__]:
+        return __calamine__
+
+    def load_workbook(self, filepath_or_buffer) -> str:
```

Comment:
This just seems to return a file name whereas the other readers return a physical workbook object. Does calamine not have a similar concept?

Reply:
No, `python-calamine` doesn't support returning data as workbook. You can use https://github.com/ToucanToco/fastexcel, if you need this feature.

Reply:
Yea I think we should follow the requirements of the ABC here - otherwise things get pretty wonky across the various implementations. How hard would it be to implement that?

Reply:
I made simple implementation of these classes (https://github.com/kostyafarber/pandas/pull/5), but I need some time to test it.

Reply:
@WillAyd I rewrote this code to use classes.